src/test/java/com/spotify/sparkey/extra/ThreadLocalWithFallbackTest.java

```diff
@@ -0,0 +1,77 @@
+package com.spotify.sparkey.extra;
+
+import org.junit.Test;
+
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ThreadLocalWithFallbackTest {
+
+  private final AtomicInteger closedThreads = new AtomicInteger();
+  private final ThreadLocalWithFallback<Long> threadLocal =
+          ThreadLocalWithFallback.withInitial(() -> Thread.currentThread().getId(), threadId -> closedThreads.incrementAndGet());
+
+  @Test
+  public void testSimple() {
+    assertEquals(1, threadLocal.get().intValue());
+  }
+
+  @Test
+  public void testSet() {
+    assertEquals(Thread.currentThread().getId(), threadLocal.get().longValue());
+    threadLocal.set(123L);
+    assertEquals(123, threadLocal.get().longValue());
+    threadLocal.remove();
+    assertEquals(Thread.currentThread().getId(), threadLocal.get().longValue());
+  }
+
+  @Test
+  public void testForkJoinPoolCommonPool() {
+    final int poolSize = ForkJoinPool.commonPool().getParallelism();
+    final AtomicInteger failures = new AtomicInteger();
+    for (int i = 0; i < 10000; i++) {
+      int finalI = i;
+      ForkJoinPool.commonPool().execute(() -> {
+        checkFailure(failures);
+        if (finalI < poolSize) {
+          try {
+            Thread.sleep(1);
```

Comment:
out of curiosity, why are you doing a `Thread.sleep` here? Trying to force it to use the whole pool?

Reply:
Yes, exactly